File: binarySearch.c
#include<stdio.h>

int insertionSort(int arry[], int n){
    int i, d, temp;
    for(i=1; i<n; i++){
        d=i;
        while(d>0 && arry[n]<arry[n-1]){
            temp=arry[n];
            arry[n]=arry[n-1];
            arry[n-1]=temp;
            d--;
        }
    }
    return arry;
}

int main(){
    int n,i, *arr;
    scanf("%d",&n);
    int arry[n];
    for(i=0; i<n; i++){
        scanf("%d",arry[i]);
    }
    
    arr = insertionSort(arry, n);

    for(i=0; i<n; i++){
        printf("*(arr + %d) : %d \n", i, *(arr+i));
    }
    
    return 0;
}

File: binarySearch.pl
#! /usr/bin/perl

sub insertionSort{
    my @array = @_;
    my $temp;
    for(my $i = 1; $i<scalar(@array); $i++){
        my $d = $i;
        while($d>0 && $array[$d]<$array[$d-1]){
            $temp = $array[$d];
            $array[$d] = $array[$d-1];
            $array[$d-1] = $temp;
            $d--;
        }
    }
    
    return @array;
}

print "Enter the number of elements";
$num = <STDIN>;

print "Enter the list of nums:\n";
my @array;

for(my $i=0; $i<$num; $i++){
    $array[$i]= <STDIN>;
} 
@sortedArray = insertionSort(@array);
print "New: @sortedArray";
